File: 2744-find-maximum-number-of-string-pairs/2744-find-maximum-number-of-string-pairs.java
class Solution {
    public int maximumNumberOfStringPairs(String[] words) {
        int maxPairs = 0;
        
        Set<String> wordSet = new HashSet<>();
        
        for(String word : words) {
            StringBuilder sb = new StringBuilder(word);
            String reversedWord = sb.reverse().toString();
            
            if(wordSet.contains(reversedWord)) {
                maxPairs++;
            }
            
            wordSet.add(word);
        }
        
        return maxPairs;
    }
}

File: 2744-find-maximum-number-of-string-pairs/2744-find-maximum-number-of-string-pairs.py
class Solution(object):
    def maximumNumberOfStringPairs(self, words):
        n = len(words)
        cnt = 0
        seen = {}
        for i in range(n):        
            if seen.get(words[i][::-1]):
                seen[words[i][::-1]] += 1
            else:
                seen[words[i]] = 1
        for key,val in seen.items():
            if val>1:
                cnt += val-1
        return cnt
        """
        :type words: List[str]
        :rtype: int
        """
        